### ISSUE TITLE
`type[type[int]]` is handled inconsistently with Pyright

### ISSUE THREAD

**Issue by kc0506** [microsoft/pylance-release#6476]
<h3> Environment data </h3>

-   Language Server version: 2024.9.2
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): 
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: basic

<h3> Code Snippet </h3>

```python
def f(a: type[type[int]]): ...

f(type)
f(type[int])
f(int)
```

<h3> Expected behavior </h3>

When checking with `pyright`, the output is:
```
./new.py:6:3 - error: Argument of type "type[int]" cannot be assigned to parameter "a" of type "type[type[int]]" in function "f"
    Type "int" is not assignable to type "type[int]" (reportArgumentType)
1 error, 0 warnings, 0 informations 
```
i.e. only `f(int)` reports error.

<h3> Actual behavior </h3>

The error reported by Pylance is inconsistent with Pyright, and the error message itself does not make much sense.

![image](https://github.com/user-attachments/assets/97bfe12a-c567-4836-bbad-0662717b9a99)
![image](https://github.com/user-attachments/assets/a02bb837-2833-4fb2-b004-7d59eb7f4c9a)

**Comment by erictraut** [comment]
This is due to a bug that was fixed in pyright 1.1.380. The latest version of pyright is 1.1.382. The version of pylance that you're using is probably based on a version of pyright prior to 1.1.380. You can switch to the prerelease version of pylance if you'd like, and that should address this.

**Comment by kc0506** [comment]
Thanks!
